MidasCivil_Adapter/PrivateHelpers/MetaData.cs

```diff
@@ -0,0 +1,273 @@
+        {
+            string path = m_directory + @"\TextFiles\00_MetaData";
+
+            if (!Directory.Exists(path))
+            {
+                Directory.CreateDirectory(path);
+            }
+
+            path += "\\" + "UNIT.txt";
+
+            if (!File.Exists(path))
+            {
+                string units = "*UNIT\n";
+
+                string[] lengths = { "M", "CM", "MM", "FT", "IN" };
+                string[] forces = { "N", "KN", "KGF", "TONF", "LBF", "KIPS" };
+                string[] temperatures = { "C", "F" };
+                string[] heats = { "CAL", "KCAL", "J", "KJ", "BTU" };
+
+                if (Array.Exists(forces, element => element == force))
+                {
+                    units += force + ", ";
+                }
+                else
+                {
+                    Engine.Reflection.Compute.RecordWarning($"Unit input {force} not recognised using Newtons (N) instead");
+                    units += "N" + ", ";
+                }
+
+                if (Array.Exists(lengths, element => element == length))
+                {
+                    units += length + ", ";
+                }
+                else
+                {
+                    Engine.Reflection.Compute.RecordWarning($"Unit input {length} not recognised using meters (m) instead");
+                    units += "M" + ", ";
+                }
+
+                if (Array.Exists(heats, element => element == heat))
+                {
+                    units += heat + ", ";
+                }
+                else
+                {
+                    Engine.Reflection.Compute.RecordWarning($"Unit input {heat} not recognised using kilojoules (kJ) instead");
+                    units += "KJ" + ", ";
+                }
+                if (Array.Exists(temperatures, element => element == temperature))
+                {
+                    units += temperature;
+                }
+                else
+                {
+                    Engine.Reflection.Compute.RecordWarning($"Unit input {temperature} not recognised using celcius (C) instead");
+                    units += "C";
+                }
+                units += "\n";
+
+                File.WriteAllText(path, units);
+            }
+
+            return true;
+        }
+
+        private bool SetVersion(string version)
+        {
+            string path = m_directory + @"\TextFiles\00_MetaData";
+
+            if (!Directory.Exists(path))
+            {
+                Directory.CreateDirectory(path);
+            }
+
+            path += "\\" + "VERSION.txt";
+
+            if (File.Exists(path))
+            {
+                Engine.Reflection.Compute.RecordWarning($"VERSION.txt already exists, version number has been updated.");
+            }
+
+            File.WriteAllText(path, $"\n*VERSION\n   {version}\n");
+
+            return true;
+        }
+
+
+        private MetaData getMetaData(string filepath)
```

Comment:
This would sit under `Read` - see this `BarReleases` example: https://github.com/BHoM/MidasCivil_Toolkit/blob/master/MidasCivil_Adapter/CRUD/Read/Properties/BarReleases.cs

You will need to add an `if` statement to `Read.cs`: https://github.com/BHoM/MidasCivil_Toolkit/blob/master/MidasCivil_Adapter/CRUD/Read/Read.cs